### PR TITLE
Vahe:  Tasks 211 220

### PR DESCRIPTION
**Pull request by hsp-vahe**

### CODE REVIEW

File: src/main/java/am/hsp/apps/seua/base/TaskArray.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * Base class for all arrays . Will contain general methods for the tasks.
+ * @author vahe
+ *
+ */
+public abstract class TaskArray extends TaskBase {
+	protected int[] getArray(int range, int arrayLength) {
+		int array[];
+		array = new int[arrayLength];
+		for(int i = 0; i < arrayLength; i++) {
+			array[i] = (int) (Math.random() * range * 2) - range;
```
[the remark this answers is not in the document]
Comment (hsp-vahe) [reply src/main/java/am/hsp/apps/seua/base/TaskArray.java]: but I can`t pass a negative value to java.util.Random class .
I red about how to do that but everywere is written about how to get negative value, nothing about how to pass.

[the remark this answers is not in the document]
Comment (hsp-harut) [reply src/main/java/am/hsp/apps/seua/base/TaskArray.java]: You need to write your custom logic, which will return the required random number from given range, by using java.util.Random.

File: src/main/java/am/hsp/apps/seua/base/TaskBase.java [hunk 2]
```diff
@@ -209,6 +211,30 @@ protected boolean isPrimeNumber(int number) {
...
+	
+	/**
+	 * Get random number from given range.
+	 * @param startRange
+	 * @param endRange
+	 * @return
+	 */
+	protected int getRandomNumberFromGivenRange(int startRange, int endRange) {
+		Random rand = new Random();
+		int randomNumber = rand.nextInt(endRange - startRange) + startRange;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/base/TaskBase.java]: The maximum value that can be retrieved from nextInt() method is Exclusive the given parameter.

File: src/main/java/am/hsp/apps/seua/tasks_211_220/Task_211.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+				midleAverage += array[i];
+			}
+		}
+		return  count == 0 ? 0 : (double) midleAverage / count;
+	}
+	
+	/**
+	 * Task test with given array.
+	 */
+	private void test() {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_211_220/Task_211.java]: No. My last comment was related to having another method which is for Printing the given array

File: src/main/java/am/hsp/apps/seua/tasks_211_220/Task_212.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		System.out.println(getMidleSquareOfPositiveElements(array));
+	}
+	
+	/**
+	 * return middle square of the array`s positive elements.
+	 */
+	private double getMidleSquareOfPositiveElements(int array[]) {
+		int midleSquare = 0;
+		int count = 0;
+		int elementPow = 0;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_211_220/Task_212.java]: 1. variables should be declared where you actually need them.
2. no need for this variable at all...

File: src/main/java/am/hsp/apps/seua/tasks_211_220/Task_213.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+		int array[] = getArray(startRange, endRange, arrayLength);
+		
+		System.out.println(getMidleSquareOfNegativeElements(array));
+		
+	}
+	
+	/**
+	 * return midle square of the array`s negative elements.
+	 */
+	private double getMidleSquareOfNegativeElements(int array[]) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_211_220/Task_213.java]: same here

File: src/main/java/am/hsp/apps/seua/tasks_211_220/Task_217.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+		
+		System.out.println(getProductSquareOddIndexOfArray(array));
+	}
+	
+	/**
+	 * return square product of odd index elements of the array.
+	 */
+	private int getProductSquareOddIndexOfArray(int array[]) {
+		int product = 1;
+		int elementPow = 0;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_211_220/Task_217.java]: same

File: src/main/java/am/hsp/apps/seua/tasks_211_220/Task_219.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+		int number = getUserInput("Please enter the number");
+		
+		System.out.println(getCountOfIndexMultiplierForNumber(number, array));
+		
+	}
+	
+	/**
+	 * Get count of index of array is multiplier for input number. 
+	 */
+	private int getCountOfIndexMultiplierForNumber(int number, int array[]) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_211_220/Task_219.java]: bad variable name: number

File: src/main/java/am/hsp/apps/seua/tasks_211_220/Task_219.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+		int count = 0;
+		for(int i = 0; i < array.length; i += number) {
+			if(isMutiplier(number, i)) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_211_220/Task_219.java]: fix this

**Pull request merged**